Software/C/GoPiGo3.cpp

```diff
@@ -9,7 +9,68 @@
  *  C++ drivers for the GoPiGo3
  */
 
-#include "GoPiGo3.h"
+#include <GoPiGo3.h>
+#include <stdexcept>
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <linux/spi/spidev.h>
+#include <stdio.h>            // for printf
+#include <string.h>           // for strstr
+#include <sys/time.h>         // for clock_gettime
+#include <unistd.h>
+
+#define FIRMWARE_VERSION_REQUIRED "0.3." // Firmware version needs to start with this
+
+#define SPI_TARGET_SPEED 500000 // SPI target speed of 500kbps
+
+#define SPIDEV_FILE_NAME "/dev/spidev0.1" // File name of SPI
+
+int spi_file_handle = -1;                    // SPI file handle
```

Comment:
While this and the following lines (28 through 73) are used for the GoPiGo3, they are not specific to the GoPiGo3, which is why I intentionally left them in the .h file.